posydon/utils/interpolators.py

```diff
@@ -39,12 +39,22 @@ def __init__(self, x, y, kind='linear', **kwargs):
                     - size 1 : use y-value for below and above
                     - size 2 : use first y-value for below and second for above
 
-        """
-        self.x = np.array(x)
-        self.y = np.array(y)
+        """ 
         if kind not in ['linear']:
             raise NotImplementedError(f"kind = {kind} is not supported")
         self.kind = kind
+
+        self.x = np.array(x)
+        self.y = np.array(y)
+        # check that the data is sorted;
+        # if not, check if it's strictly decreasing and flip arrays accordingly
+        if not np.all(np.diff(self.x) > 0):
+            if np.all(np.diff(self.x) < 0):
+                self.x = np.flip(self.x)
+                self.y = np.flip(self.y)
+            else:
+                raise ValueError("x values must be strictly increasing or strictly decreasing.")
+        
```

Comment:
Some small style changes:
```suggestion
        self.x = np.array(x)
        self.y = np.array(y)
        # check that x is increasing
        if not np.all(np.diff(self.x) > 0):
            # if instead being strictly decreasing, flip data
            if np.all(np.diff(self.x) < 0):
                self.x = np.flip(self.x)
                self.y = np.flip(self.y)
            else:
                raise ValueError("x values must be strictly increasing or "
                                 "strictly decreasing.")
```